dvc/stage/cache.py

```diff
@@ -226,6 +227,9 @@ def transfer(self, from_odb, to_odb):
 
         for src in from_fs.find(runs):
             rel = from_fs.path.relpath(src, from_odb.fs_path)
+            if not isinstance(to_fs, LocalFileSystem):
+                rel = from_fs.path.as_posix(rel)
```

Comment:
@efiop, this shows that we really need to use posix paths everywhere.

Reply:
Hopefully this should be mitigated with run-cache object introduction.